components/forms/EmailForm.tsx

```diff
@@ -10,20 +11,36 @@ interface EmailFormProps {
 export const EmailForm = (props: EmailFormProps) => {
   const [email, setEmail] = useState('')
   const [isEntering, setIsEntering] = useState(false)
+  const [isSuccessOpen, setIsSuccessOpen] = useState(false);
+  const [isErrorOpen, setIsErrorOpen] = useState(false);
+
+  const handleSuccessModal = () => {
+    setIsSuccessOpen(true);
+};
+
+const handleCloseModal = () => {
+    setIsSuccessOpen(false);
+};
+
+const handleOpenErrorModal = () => {
+  setIsErrorOpen(true);
+};
+
+const handleCloseErrorModal = () => {
+  setIsErrorOpen(false);
+};
 
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault()
     addToMailchimp(email)
       .then((data: any) => {
-        alert(data.msg)
+        handleOpenErrorModal();
```

Comment:
this is only ever going to open the error modal!

Reply:
Actually the error is caught first before the error modal. For correctly formatted emails it will never open the Error Modal

Reply:
But on success it seems to be calling `handleOpenErrorModal();` and not `handleSuccessModal()` 
Either way, lets make this a little nicer with the below suggestion